### PR TITLE
feat: EIP-3009 support

### LINKED ISSUE #12: EIP-3009 Support
Adding EIP-3009 support to USDFC will allow for its use in Coinbase's [x402.org](https://x402.org) protocol.

### PR DESCRIPTION
**Pull request by lanzafame**
## Summary
- Implemented EIP-3009 (Transfer With Authorization) support for DebtToken
- Added comprehensive test suite for EIP-3009 functionality
- Updated interfaces and borrower operations to support the new feature

## Test plan

Note: tests exist but some are failing, I believe, due to the EIP-3009 changes not being deployed on testnet, please let me know what the actual deal is 🙏 

- [x] Unit tests for basic EIP-3009 functions (receiveWithAuthorization, transferWithAuthorization, cancelAuthorization)
- [x] Integration tests with existing protocol functionality
- [x] Test coverage for edge cases and security considerations
- [x] Verify backward compatibility with existing DebtToken functionality

Closes #12

### CODE REVIEW

File: contracts/DebtToken.sol [hunk 1]
```diff
@@ -336,4 +351,136 @@ contract DebtToken is OwnableUpgradeable, CheckContract, IDebtToken {
...
+        address recoveredAddress = ecrecover(digest, _v, _r, _s);
+        require(
+            recoveredAddress != address(0) && recoveredAddress == _from,
+            "DebtToken: invalid signature"
+        );
+
+        _authorizationStates[_from][_nonce] = true;
+        emit AuthorizationUsed(_from, _nonce);
+
+        _requireValidRecipient(_to);
```
[the remark this answers is not in the document]
Comment (biga816) [reply contracts/DebtToken.sol]: Could you fix `receiveWithAuthorization` as well?

**Review by biga816 — APPROVED** [review]
